README.md

```diff
@@ -16,6 +16,92 @@ See [PROPOSAL.md](PROPOSAL.md) for the proposal for this project.
 
 # Documentation
 
+## Overview
+
+What if you could just push a change into a git repository and have everything from tests to deployment in a cluster run automatically without any needed supervision? Our project shows how to do exactly that.
+
+Using an Angular demo app called demo-app that simply displays something and has some tests, we use Concourse (an open source CI/CD system) to run the tests, and if they succeed upload a new docker image of the demo app to Docker Hub. Finally, an operator watches over the demo app deployment to make sure the demo app and its service are deployed and updates them if a new version on Docker Hub is available.
+
+All of the above (demo app, Concourse and the operator) run on a kubernetes cluster. All images are from (or through) Docker Hub.
+
+## Demo App
+
+This is just a demo app built with Angular. It displays some text. It has tests.
+
+The [demo-app-statefulset.yaml](demo-app-statefulset.yaml) configuration defines a stateful set with three pods and a service with type NodePort.
+
+We use a stateful set instead of a replica set because of the nice predictable and readable names (e.g. demo-app-0, demo-app-1 and demo-app-2) and the service type NodePort because the cluster we deploy on has an Nginx reverse proxy.
+
+## Concourse
+
+Concourse is a CI/CD system which allows creating pipelines consisting of jobs, where each job runs in its own container. We use Concourse mainly because it is a cool project and does what we want: running jobs (which includes tests, of course).
```

Comment:
```suggestion
Concourse is a CI/CD system which allows creating pipelines consisting of jobs.
Each job runs tasks which run, e.g., scripts, in their own containers. 
We chose Concourse because it is a common CI/CD system, uses YAML to configure its pipelines, and does what it should: running automated jobs (which includes tests, of course).
```